jupyter/d2l-java/chapter_multilayer-perceptrons/mlp-djl.ipynb

```diff
@@ -0,0 +1,323 @@
+    "of softmax regression implementation\n",
+    "(:numref:`sec_softmax_gluon`),\n",
+    "the only difference is that we add \n",
+    "*two* `Linear` (fully-connected) layers \n",
+    "(previously, we added *one*).\n",
+    "The first is our hidden layer, \n",
+    "which contains *256* hidden units\n",
+    "and applies the ReLU activation function.\n",
+    "The second is our output layer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "5"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "SequentialBlock net = new SequentialBlock();\n",
+    "net.add(Blocks.batchFlattenBlock(784));\n",
+    "net.add(Linear.builder().setOutChannels(256).build());\n",
+    "net.add(Activation::relu);\n",
+    "net.add(Linear.builder().setOutChannels(10).build());\n",
+    "net.setInitializer(new NormalInitializer());"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that DJL, as usual, automatically\n",
+    "infers the missing input dimensions to each layer.\n",
+    "\n",
+    "The training loop is *exactly* the same\n",
+    "as when we implemented softmax regression.\n",
+    "This modularity enables us to separate \n",
+    "matters concerning the model architecture\n",
+    "from orthogonal considerations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "int batchSize = 256;\n",
+    "int numEpochs = 10;\n",
+    "double[] trainLoss;\n",
+    "double[] testAccuracy;\n",
+    "double[] epochCount;\n",
+    "double[] trainAccuracy;\n",
+    "\n",
+    "trainLoss = new double[numEpochs];\n",
+    "trainAccuracy = new double[numEpochs];\n",
+    "testAccuracy = new double[numEpochs];\n",
+    "epochCount = new double[numEpochs];\n",
+    "\n",
+    "FashionMnist trainIter = FashionMnist.builder()\n",
+    "                            .optUsage(Dataset.Usage.TRAIN)\n",
+    "                            .setSampling(batchSize, true)\n",
+    "                            .build();\n",
+    "\n",
+    "\n",
+    "FashionMnist testIter = FashionMnist.builder()\n",
+    "                            .optUsage(Dataset.Usage.TEST)\n",
+    "                            .setSampling(batchSize, true)\n",
+    "                            .build();\n",
+    "\n",
+    "trainIter.prepare();\n",
+    "testIter.prepare();\n",
+    "\n",
+    "for(int i = 0; i < epochCount.length; i++) {\n",
+    "    epochCount[i] = (i + 1);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "// function to collect results from metrics.\n",
+    "public void collectResults(Trainer trainer, Evaluator evaluator) {\n",
+    "    Metrics metrics = trainer.getMetrics();\n",
+    "    boolean hasTrainMetric = metrics.hasMetric(EvaluatorTrainingListener\n",
```

Comment:
We don't need this function outside of this book/script. Does hasTrainMetric vary within the book?